src/components/anchor-navigation/anchor-navigation.test.tsx

```diff
@@ -0,0 +1,364 @@
+  (scrollPosition, selectedAnchorIndex) => {
+    renderComponent();
+
+    const topEdgeOffsets = [400, 800, 1200, 1600, 2000];
+    const refs = [ref1, ref2, ref3, ref4, ref5];
+
+    refs.forEach((ref, index) => {
+      if (ref.current) {
+        jest
+          .spyOn(ref.current, "getBoundingClientRect")
+          .mockImplementation(
+            () => ({ top: topEdgeOffsets[index] - scrollPosition } as DOMRect)
+          );
+      }
+    });
+
+    jest
+      .spyOn(screen.getByRole("list"), "getBoundingClientRect")
+      .mockImplementation(
+        () => ({ top: SECTION_VISIBILITY_OFFSET } as DOMRect)
+      );
+
+    act(() => {
+      window.dispatchEvent(new Event("scroll"));
+    });
+    act(() => {
+      jest.advanceTimersByTime(10);
+    });
+
+    SELECTED_ITEM_STYLES.forEach(([property, value]) => {
+      expect(
+        screen.getAllByRole("listitem")[selectedAnchorIndex]
+      ).toHaveStyleRule(property, value, { modifier: "& a" });
+    });
+  }
+);
+
+test("cleans up event listeners after unmounting", () => {
+  const { unmount } = renderComponent();
+  const addEventListenerSpy = jest.spyOn(window, "removeEventListener");
+
+  unmount();
+
+  expect(
+    addEventListenerSpy.mock.calls.filter((call) => call[0] === "scroll")
+  ).toHaveLength(1);
+});
+
+describe("validates incorrect stickyNavigation prop content", () => {
+  let mockGlobal: jest.SpyInstance;
+
+  beforeEach(() => {
+    mockGlobal = jest
+      .spyOn(global.console, "error")
+      .mockImplementation(() => undefined);
+  });
+
+  afterEach(() => {
+    mockGlobal.mockReset();
+  });
+
+  test("items that are not AnchorNavigationItems", () => {
+    const error = `\`stickyNavigation\` prop in \`AnchorNavigation\` should be a React Fragment that only contains children of type \`${AnchorNavigationItem.displayName}\``;
+
+    expect(() => {
+      render(
+        <AnchorNavigation
+          stickyNavigation={
+            <>
+              <p>Invalid children</p>
+            </>
+          }
+        />
+      );
+    }).toThrow(error);
+  });
+
+  test("container that is not a React Fragment", () => {
+    const error =
+      "`stickyNavigation` prop in `AnchorNavigation` should be a React Fragment.";
+
+    expect(() => {
+      render(
+        <AnchorNavigation
+          stickyNavigation={
+            <div>
+              <AnchorNavigationItem>First</AnchorNavigationItem>
+            </div>
+          }
+        />
+      );
+    }).toThrow(error);
+  });
+});
+
+test("renders not selected navigation item with proper background when hovered", async () => {
+  renderComponent();
+
+  expect(screen.getAllByRole("listitem")[1]).toHaveStyleRule(
+    "background-color",
+    "var(--colorsActionMinor100)",
+    { modifier: "& a:hover" }
+  );
+});
+
+test("renders selected navigation item with proper background when hovered", async () => {
```

Comment:
**question**: are these, and other styling related tests included to meet coverage?

Reply:
An equivalent test was there in the Enzyme tests, and I hadn't thought too much about it as it wasn't too hard to convert this one to RTL. I don't know if it's needed for coverage - I doubt it to be honest - but I quickly checked the Playwright tests and they don't appear to test anything to do with hovering, so we should have this in imo. It might be better to add this in playwright, but given that this was in the unit tests before and the time pressure we're under I'd prefer to leave these in, we can always decide to change tests later after a thorough review of what should be tested in each place. (Particularly if we can get combined coverage working.)